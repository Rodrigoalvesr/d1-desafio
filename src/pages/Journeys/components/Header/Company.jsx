import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import { makeStyles } from '@material-ui/core/styles';
import Tooltip from '@Components/Tooltip';
import companyLogo from '@/assets/images/acme-logo.png';
import { withStyles } from '@material-ui/core/styles';

const DarkTooltip = withStyles({
  tooltip: {
    color: '#FFFFFF',
    backgroundColor: '#3E4157',
  },
  arrow: {
    color: '#3E4157',
  },
})(Tooltip);

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
  },
  avatar: {
    width: 24,
    height: 24,
    fontSize: 14,
    fontWeight: 'bold',
    background: theme.palette.primary.main,
  },
  avatarContent: {
    marginRight: 15,
  },
  companyContent: {
    display: 'flex',
    alignItems: 'center',
    height: 30,
    padding: '0 13px',
    border: '1px solid #EBEEF6',
    borderRadius: 5,
    background: '#FFFFFF',
  },
}));

function Company() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <div className={classes.avatarContent}>
        <DarkTooltip title="AntÃ´nio da Silva" placement="right" arrow>
          <Avatar className={classes.avatar}>A</Avatar>
        </DarkTooltip>
      </div>
      <div className={classes.companyContent}>
        <img alt="acme" src={companyLogo} height={26} />
      </div>
    </div>
  );
}

export default Company;
