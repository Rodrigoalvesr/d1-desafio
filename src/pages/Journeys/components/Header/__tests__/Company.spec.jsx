import React from 'react';
import renderer from 'react-test-renderer';
import Company from '../Company';

describe('Company', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Company />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
