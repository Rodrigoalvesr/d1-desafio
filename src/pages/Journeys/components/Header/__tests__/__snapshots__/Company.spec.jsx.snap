// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-avatarContent-3"
  >
    <div
      aria-describedby={null}
      className="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-2 MuiAvatar-colorDefault"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="AntÃ´nio da Silva"
    >
      A
    </div>
  </div>
  <div
    className="makeStyles-companyContent-4"
  >
    <img
      alt="acme"
      height={26}
      src="acme-logo.png"
    />
  </div>
</div>
`;
